pkg/util/util.go

```diff
@@ -1173,6 +1176,21 @@ func GetAWSRegion() (string, error) {
 		return "", fmt.Errorf("Unexpected node name format: %q", nodesList.Items[0].Name)
 	}
 	awsRegion := mapValidAWSRegions[nameSplit[1]]
+
+	// Fetching aws region from cluster if not fetched from node name
+	if awsRegion == "" {
```

Comment:
@achouhan09, using the infrastructure CR looks like a better way to get the region than parsing the node name. Maybe we should first attempt to get it this way, and only fallback to the node name if we fail. WDYT?

Reply:
Yes, make sense. I have updated the code. Thanks